---
title: "Genetic Variation among Channel Island Foxes"
output: pdf_document
date: "2022-11-11"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(ape)

```

```{r fig.width=28, fig.height=12}
# island fox mitochondrial genome tree

phy = read.tree("islandmito.fasta.treefile")
plot(phy)

```

```{r}
library(ggplot2)
library(adegenet)
library(ade4)

snp <- fasta2genlight('islandmito.fasta', snpOnly=T)


position(snp)
alleles(snp)
locNames(snp)

## plot positions of polymorphic sites
temp <- density(position(snp), bw=10)
plot(temp, xlab="Position in the alignment", lwd=2, main="Location of the SNPs")
points(position(snp), rep(0, nLoc(snp)), pch="|", col="red")

polygon(c(temp$x,rev(temp$x)), c(temp$y, rep(0,length(temp$x))),
col=transp("blue",.3))
points(position(snp), rep(0, nLoc(snp)), pch="|", col="blue")

```

```{r}
#density by codon position
snpposi.plot(position(snp), genome.size=1700, codon=FALSE)

glPlot(snp, posi="topleft")

myFreq <- glMean(snp)
hist(myFreq, proba=TRUE, col="gold", xlab="Allele frequencies",
main="Distribution of (second) allele frequencies")
temp <- density(myFreq)
lines(temp$x, temp$y*1.8,lwd=3)

myFreq <- glMean(snp)
myFreq <- c(myFreq, 1-myFreq)
hist(myFreq, proba=TRUE, col="darkseagreen3", xlab="Allele frequencies",
main="Distribution of allele frequencies", nclass=20)
temp <- density(myFreq, bw=.05)
lines(temp$x, temp$y*2,lwd=3)

#snpposi.plot(position(snp), genome.size=1700)

#as.randtest(sim = sim, obs = snp, alter = "less")
#temp <- table(unlist(other(snp)))
#barplot(temp, main="Distribution of the number \nof alleles per loci",
#xlab="Number of alleles", ylab="Number of sites", col=heat.colors(4))

```

```{r}
#pca1 <- prcomp(position(snp), scale = TRUE)
pca1 <- glPca(snp)
scatter(pca1, posi="bottomright")

library(ape)
tre <- nj(dist(as.matrix(snp)))
tre

#plot(tre, typ="fan", cex=1)

#myCol <- colorplot(pca1$scores,pca1$scores, transp=TRUE, cex=4)
#abline(h=0,v=0, col="grey")
#add.scatter.eig(pca1$eig[1:40],2,1,2, posi="topright", inset=.05, ratio=.3)


```

```{r}
ggplot(snp, aes(x=temp, alpha=0.5))+geom_density()+theme_bw()


library(devtools)
library(dartR)

gl.install.vanilla.dartR()


pc <- gl.pcoa(gl, nfactors=5)




```

```{r}
meta = read.table('meta.csv', sep=',', header = T)
pca <- glPca(snp, nf = 10)
barplot(100*pca$eig/sum(pca$eig), main="Eigenvalues", col=heat.colors(length(pca$eig)))
scatter(pca, psi='bottomright')
glPlot(snp)



#library(factoextra)
#res.pca <- prcomp(meta.csv, scale = TRUE)
#fviz_pca_ind(res.pca,
             #col.ind = "cos2", # Color by the quality of representation
            # gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07"),
            # repel = TRUE     # Avoid text overlapping
           #  )


#pca.dataset = as.data.frame(pca$scores)
#pca.dataset$isolates = rownames(pca.dataset)
#pca.dataset$pop = as.factor(meta[match(snp$ind.names, meta$Strain),]$SamplingSite)
#pca.dataset$spp = as.factor(meta[match(snp$ind.names, meta$Strain),]$Species)

#ggplot(pca.dataset, aes(PC1, PC2, fill=spp)) + geom_point(shape=21, size=3, alpha=0.7)
#ggplot(pca.dataset, aes(PC1, PC3, fill=spp)) + geom_point(shape=21, size=3, alpha=0.7)


```

```{r}
#Raw_Data = read.csv('meta.csv')
#Data_of_Int = Raw_Data[3:7]
#PRcomp_Stand <- prcomp(Raw_Data,scale=T)
#fviz_eig(PRcomp_Stand)

#fviz_pca_var(PRcomp_Stand,
            # col.var = "contrib", # Color by contributions to the PC
          #   gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07"),
           #  repel = TRUE     # Avoid text overlapping
#)


#fviz_pca_ind(PRcomp_Stand,
         #    col.ind = "cos2", # Color by the quality of representation
         #    gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07"),
          #   repel = TRUE     # Avoid text overlapping
#)


```
